### PR TITLE
Add beta support of Safari

### PR DESCRIPTION
**Pull request by hobinjk**
Note that this uses a copy of all extension files instead of using a build step to copy them. This also uses a squashed version of the commits, I can unsquash if desired.

### CODE REVIEW

File: New XKit.safariextension/Info.plist [hunk 1]
```diff
@@ -3,23 +3,23 @@
...
+	<string>New XKit Team</string>
 	<key>Builder Version</key>
-	<string>10600.1.25</string>
+	<string>10601.1.56.2</string>
 	<key>CFBundleDisplayName</key>
-	<string>XKit</string>
+	<string>New XKit</string>
 	<key>CFBundleIdentifier</key>
-	<string>com.studioxenix.xkit</string>
+	<string>com.xkitplus.xkit</string>
```
Comment (0xazure) [inline New XKit.safariextension/Info.plist]: `xkitplus` vs `new[-]xkit`?

**Comment by nightpool** [comment]
Unsquashd would probably be better for commit history purposes.

Are there any safari-specific diffs or are these straight copies?

**Comment by 0xazure** [comment]
It's probably safe to remove the extension file copies, unless you want to leave them here and remove them in whatever PR closes #546.

Also, the XKit Editor doesn't seem to load on the Safari version and I'm not entirely clear why.  I didn't notice any obvious error messages in the console, but it might have something to do with the Ace editor?

**Comment by Wolvan** [comment]
@0xazure mind if I take a crack at the build process?

**Comment by hobinjk** [comment]
@nightpool There are not actually that many changes, I think I will leave it squashed for now. It's only Rewrite global.js, Fix Whitespace Errors, Update/Fix Info.plist and then some miscellaneous bugfixes.

@0xazure I agree, I've removed the copies and renamed the folder.

File: Safari/Info.plist [hunk 2]
```diff
@@ -3,23 +3,23 @@
...
+	<string>New XKit Team</string>
 	<key>Builder Version</key>
-	<string>10600.1.25</string>
+	<string>10601.1.56.2</string>
 	<key>CFBundleDisplayName</key>
-	<string>XKit</string>
+	<string>New XKit</string>
 	<key>CFBundleIdentifier</key>
-	<string>com.studioxenix.xkit</string>
+	<string>com.xkitplus.xkit</string>
```
Comment (0xazure) [inline Safari/Info.plist]: Rebase made GH hid my comment, so here it is again:

`xkitplus` vs `new[-]xkit`?

Comment (hobinjk) [inline Safari/Info.plist]: Oh right, sorry for not addressing this. I don't think we own new-xkit.com or newxkit.com, but someone has xkitplus.com. That is the rational for this identifier.

Comment (0xazure) [inline Safari/Info.plist]: @ChuckL has https://portal.new-xkit.com/#/login for the XCloud login.

Comment (Wolvan) [inline Safari/Info.plist]: We, or rather @ChuckL, does in fact own new-xkit.com

Comment (hobinjk) [inline Safari/Info.plist]: Great, pointing it there.

**Comment by 0xazure** [comment]
I'm still getting issues with

```sh
XBackground Error: TypeError: undefined is not an object (evaluating 'ev.target.page.dispatchMessage')
```

on certain blogs.  This is a major blocker for us to release the Safari version, especially since it produces a whole chain of pop-up error messages.

**Comment by 0xazure** [comment]
In attempting to test out #571 I've been running into a lot of 

```sh
XBridge: Warning! Callback for HTTP Request lIQ5Ox1fICmjfIdC608eUs0EpenzGFBdwrg4e2tObd3cqB9CxE not found!
```

errors.  I'm not _quite_ sure that this means, but I do know that I cannot properly update any of the XKit extensions.  XKit barely installs itself correctly most of the time.  

I've set XKit to connect locally to `gulp server`, but it keeps having a `Server error, falling back for download of xkit_preferences.json`, and so falls back to the outdated distribution.  Near as I can tell this is a server error at the `xmlhttpRequest` level and not a 'server error' as identified by XKit itself, but there's no difference between the setup for Safari and the one that works on Chrome so I'm not sure what's causing the error.  I have, of course, verified that the self-signed cert is accepted by Safari and is in the cert store.

**Comment by hobinjk** [comment]
Updated with more functionality, I no longer see errors on the discussed page, but I also didn't specifically try to fix those errors.

**Comment by hobinjk** [comment]
https://files.gitter.im/new-xkit/XKit/taHc/New-XKit.safariextz is the latest build

**Comment by nightpool** [comment]
lol at sharing files over gitter haha

**Comment by 0xazure** [comment]
I'll review this when I have time, hopefully late tomorrow.  I'll take another stab at serving stuff locally and see if I can narrow the issue there down further, if it isn't already resolved.

**Comment by 0xazure** [comment]
I can confirm that the issue with

```sh
'ev.target.page.dispatchMessage'
```

has been resolved and Safari appears to be working as intended.

However, I am still unable to test extensions on Safari when serving them locally.  As can be seen below, 
`SafariExtensionMessageEvent` is returning a `status` of 0, even though the local server is reporting it served the requested asset with `status` 200.

```sh
{request: Object, status: 0, settings: undefined, request_id: "b5paRapSkhec1lakt5vaPIsWPmuB0YtIYDgQyaPYx2mlmTxl91", headers: ""}
```

This explains all of the 

```
XBridge: Warning! Callback for HTTP Request lIQ5Ox1fICmjfIdC608eUs0EpenzGFBdwrg4e2tObd3cqB9CxE not found!
```

messages, as the returned message object has no callbacks to be called.

Note that everything is working as expected when Safari sends requests to our `gh-pages` URL, so this seems to be an issue with SSL certificates even though my Safari installation is correctly configured to trust the dev cert as can be seen below:

screen shot 2015-10-12 at 20 17 44

I am of the opinion that we can track down the issue(s) with serving assets locally at a later date, so I feel comfortable merging this now.

@hobinjk anything more you want to do with this branch before I merge?

**Comment by hobinjk** [comment]
Great! I also could not figure out how to serve locally. I would guess that it is some form of extension security nonsense. Merge whenever you want to.

**Comment by 0xazure** [comment]
@homu r+

**Pull request merged**